src/hooks/useFirestoreConnect.ts

```diff
@@ -11,17 +12,36 @@ import { ValidFirestoreKeys } from "types/Firestore";
  */
 export type Defined<T> = T & Exclude<T, undefined>;
 
+/**
+ * This type allows us to avoid bugs when doc is not specified, and
+ * automagically constrain the storeAs parameter to keys that exist
+ * within ValidFirestoreKeys, ensuring that we can't forget to define
+ * it in our types when using functions that rely on this type
+ * (eg. useFirestoreConnect)
```

Comment:
This doesn't make as much sense anymore now that we're renaming this helper + shadowing `useFirestoreConnect`

Reply:
Discussed offline, cool to go with it for now.